### DISCUSSION TITLE
Taskcluster question

### DISCUSSION THREAD

**Discussion by gterzian** [servo/servo#25813] [General] [answered]
Is there an easy way to lookup a task that was started by bors but whose result failed to be reported on the PR? 

Would it be possible for the bot the post a link to the task when it's started?

**Comment by SimonSapin** [comment, marked as answer]
**TL;DR:** the link is already there, though a bit lost in the noise. Click on the green/orange/red icon for the merge commit as shown in the screenshot below.

Or, you can also find every task through Treeherder:

* https://treeherder.mozilla.org/#/jobs?repo=servo-auto
* https://treeherder.mozilla.org/#/jobs?repo=servo-try
* https://treeherder.mozilla.org/#/jobs?repo=servo-master

**Explanation:**

Let’s take for example this command given to Homu: https://github.com/servo/servo/pull/25793#issuecomment-588297491

Upon receiving this command, Homu makes a new merge commit of the PR’s "head" commit with the current `master` (here 8913c6734cf2f7c7cd59f3bb1418dbec409090fd) and pushes that to the `auto` branch. Because the message of this merge commit includes the PR number, this push is visible in the PR as a "bors-servo added a commit that referenced this pull request" event.

Next, Taskcluster reacts to this GitHub Push Event by scheduling tasks based on [`.taskcluster.yml`](https://github.com/servo/servo/blob/master/.taskcluster.yml) in the pushed branch. It also uses the GitHub Statuses API to signal this. A Status is associated to a commit SHA, not a branch. When reacting to push events TC sets the Status for the "head" commit on the push, which in this case is the merge commit created by Homu. The Status is initially Pending, and TC changes it to Success if all tasks in the task group succeed or to Error if any task errors.

Finally, Homu watches for this Status change to decide when to move on to the next PR in its queue after optionally pushing this existing merge commit to `master`. But in the meantime (and afterwards) the Status is shown on GitHub’s web UI as **an orange disc, green check mark, or red cross next to its commit** every time a commit is shown. This includes https://github.com/servo/servo/commits/auto for example, but also that "bors-servo added a commit that referenced this pull request" event.

![image](https://user-images.githubusercontent.com/291359/74906843-6dd23c00-53b2-11ea-9b81-863a8296ec43.png)

Clicking that icon shows each Status with a Details link that for Taskcluster goes to the task group:

![image](https://user-images.githubusercontent.com/291359/74906901-8d696480-53b2-11ea-965c-62aafbddb5bd.png)

-----


When a Status is set for a commit that happens to be the head commit of an open PR, GitHub shows it much more visibly in the PR. In this repository this happens **only for pre-review testing that starts as soon as you open or update a PR** (which currently only runs tidy and does a Linux dev build), not anything initiated by Homu. This is because the `auto` branch and the merge commit made by Homu are unrelated to the PR as far as GitHub knows, other than mentioning the PR number in the merge commit message.

![image](https://user-images.githubusercontent.com/291359/74907203-3748f100-53b3-11ea-9d97-0645420f86bc.png)

**Comment by gterzian** [comment]
Thanks for the info!

**Comment by gterzian** [comment]
![Screen Shot 2020-02-25 at 3 11 44 PM](https://user-images.githubusercontent.com/2792687/75223400-572c4a80-57e1-11ea-8514-c94debe88e2d.png)

Really enjoying checking the status of a task while it's ongoing! :)
